src/util/message.h

```diff
@@ -33,6 +34,12 @@ class message_handlert
     // no-op by default
   }
 
+  /// Return the underlying JSON stream
+  virtual json_stream_arrayt &get_json_stream()
+  {
+    UNREACHABLE;
+  }
```

Comment:
If I change it to `get_json_stream() = 0;` then I get compilation errors that some variables, such as  `message_handler` in line 189 of `graphml.cpp` can not be initialised to an abstract type. I understand that this happens because other members of the hierarchy, like `null_message_handlert` are not redefining `get_json_stream()`, a now pure virtual function.

If it's marked `protected`, that creates problems with the interaction with the class `mstreamt` in the same file (`message.h`, line 184), particularly with the method `json_stream()` in line 244, which contains the statement `return message.message_handler->get_json_stream();`, which can not work with `get_json_stream()` being protected.

Reply:
> If I change it to `get_json_stream() = 0;` then I get compilation errors that some variables, such as `message_handler` in line 189 of `graphml.cpp` can not be initialised to an abstract type. I understand that this happens because other members of the hierarchy, like `null_message_handlert` are not redefining `get_json_stream()`, a now pure virtual function.

To me, this looks like a prime example of why compile-time errors are preferable over runtime errors: in my opinion, that just means that `null_message_handlert` needs to be fixed.